server/src/main/java/com/aws/greengrass/cli/CLIEventStreamAgent.java

```diff
@@ -613,6 +616,8 @@ public ListLocalDeploymentsResponse handleRequest(ListLocalDeploymentsRequest re
                         persistedDeployments.add(localDeployment);
                     });
                 }
+
+                cleanUpQueuedDeployments(cliServiceConfig);
```

Comment:
doesn't this need to be called _before_ L608? The returned list of deployments would be wrong, until this cleanup happens.